src/Uno.UI/Microsoft/UI/Xaml/Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,1322 @@
+				{
+					var rightContentPresenter = m_rightContentPresenter;
+					if (rightContentPresenter != null)
+					{
+						var rightContentSize = rightContentPresenter.DesiredSize;
+						rightContentColumn.MinWidth = rightContentSize.Width;
+						widthTaken += rightContentSize.Width;
+					}
+				}
+
+				var tabColumn = m_tabColumn;
+				if (tabColumn != null)
+				{
+					// Note: can be infinite
+					var availableWidth = previousAvailableSize.Width - widthTaken;
+
+					// Size can be 0 when window is first created; in that case, skip calculations; we'll get a new size soon
+					if (availableWidth > 0)
+					{
+						if (TabWidthMode == TabViewWidthMode.Equal)
+						{
+
+							var minTabWidth = (double)SharedHelpers.FindInApplicationResources(c_tabViewItemMinWidthName, c_tabMinimumWidth);
+							var maxTabWidth = (double)SharedHelpers.FindInApplicationResources(c_tabViewItemMaxWidthName, c_tabMaximumWidth);
+
+							// If we should fill all of the available space, use scrollviewer dimensions
+							var padding = Padding;
+							if (fillAllAvailableSpace)
+							{
+								// Calculate the proportional width of each tab given the width of the ScrollViewer.
+								var tabWidthForScroller = (availableWidth - (padding.Left + padding.Right)) / (double)(TabItems.Count);
+								tabWidth = MathEx.Clamp(tabWidthForScroller, minTabWidth, maxTabWidth);
+							}
+							else
+							{
+								double availableTabViewSpace = (tabColumn.ActualWidth - (padding.Left + padding.Right));
+								var increaseButton = m_scrollIncreaseButton;
+								if (increaseButton != null)
+								{
+									if (increaseButton.Visibility == Visibility.Visible)
+									{
+										availableTabViewSpace -= increaseButton.ActualWidth;
+									}
+								}
+
+								var decreaseButton = m_scrollDecreaseButton;
+								if (decreaseButton != null)
+								{
+									if (decreaseButton.Visibility == Visibility.Visible)
+									{
+										availableTabViewSpace -= decreaseButton.ActualWidth;
+									}
+								}
+
+								// Use current size to update items to fill the currently occupied space
+								tabWidth = availableTabViewSpace / (double)(TabItems.Count);
+							}
+
+
+							// Size tab column to needed size
+							tabColumn.MaxWidth = availableWidth;
+							var requiredWidth = tabWidth * TabItems.Count;
+							if (requiredWidth >= availableWidth)
+							{
+								tabColumn.Width = GridLengthHelper.FromPixels(availableWidth);
+								var listview = m_listView;
+								if (listview != null)
+								{
+									ScrollViewer.SetHorizontalScrollBarVisibility(listview, ScrollBarVisibility.Visible);
+									UpdateScrollViewerDecreaseAndIncreaseButtonsViewState();
+								}
+							}
+							else
+							{
+								//TODO: Uno specific workaround - ListView stretches to full available width, even when it does not require it
+								//tabColumn.Width = GridLengthHelper.FromValueAndType(1.0, GridUnitType.Auto);
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6063676)